specification/monitor/resource-manager/Microsoft.Insights/stable/2024-03-11/dataCollectionRules_API.json

```diff
@@ -0,0 +1,2379 @@
+          "type": "string",
+          "example": "myAI"
+        },
+        "name": {
+          "description": "A friendly name for the data source.\r\nThis name should be unique across all data sources (regardless of type) within the data collection rule.",
+          "type": "string",
+          "example": "myDataSource1"
+        }
+      }
+    },
+    "OtelTracesDataSource": {
+      "description": "Enables Otel Traces to be collected by this data collection rule.",
+      "required": [
+        "streams"
+      ],
+      "type": "object",
+      "properties": {
+        "streams": {
+          "description": "List of streams that this data source will be sent to.\r\nA stream indicates what schema will be used for this data and usually what table in Log Analytics the data will be sent to.",
+          "type": "array",
+          "items": {
+            "enum": [
+              "Microsoft-OTel-Traces-Spans",
+              "Microsoft-OTel-Traces-Events",
+              "Microsoft-OTel-Traces-Resources"
+            ],
+            "type": "string",
+            "x-ms-enum": {
+              "name": "KnownOtelTracesDataSourceStreams",
+              "modelAsString": true
+            }
+          },
+          "example": [
+            "Microsoft-OtelSpans",
+            "Microsoft-OtelEvents",
+            "Microsoft-OtelResources"
+          ]
+        },
+        "resourceAttributeRouting": {
+          "description": "Specifies the routing policy based on OTLP payload resource attributes to route subset of the payload according to matching resource attribute.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/OtelDataSourceResourceAttributeRouting"
+            }
+          ]
+        },
+        "enrichWithResourceAttributes": {
+          "description": "Specifies the list of resource attributes that need to be added as labels/dimensions to the log record for further enrichment.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "example": [
+            "service.name"
+          ]
+        },
+        "enrichWithReference": {
+          "description": "Specifies the reference to enrich the telemetry signal with.",
+          "type": "string",
+          "example": "myAI"
+        },
+        "replaceResourceIdWithReference": {
+          "description": "Specifies whether to replace the default resourceId in the log record with the resourceId of the referenced resource being used for enrichment.",
+          "type": "boolean",
+          "example": true
+        },
+        "name": {
+          "description": "A friendly name for the data source.\r\nThis name should be unique across all data sources (regardless of type) within the data collection rule.",
+          "type": "string",
+          "example": "myDataSource1"
+        }
+      }
+    },
+    "OtelTracesDirectDataSource": {
```

Comment:
>OtelTracesDirectDataSource [](http://example.com/codeflow?start=5&length=26)

Earlier comments on similarly/identically-named properties apply to this definition as well.